compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/MethodGen.java

```diff
@@ -469,19 +470,23 @@ void generateBasicBlocks(MethodVisitor mv, LabelGenerator labelGen, JvmErrorGen
                     .getLastScopeFromBBInsGen(mv, labelGen, instGen, localVarOffset, funcName, bb,
                                               visitedScopesSet, lastScope);
 
-            Label bbEndLabel = labelGen.getLabel(funcName + bb.id.value + "beforeTerm");
-            mv.visitLabel(bbEndLabel);
 
             BIRTerminator terminator = bb.terminator;
             pushShort(mv, stateVarIndex, caseIndex);
             caseIndex += 1;
 
             processTerminator(mv, func, module, funcName, terminator, jvmTypeGen, localVarOffset);
+
             termGen.genTerminator(terminator, moduleClassName, func, funcName, localVarOffset,
                                   returnVarRefIndex, attachedType);
 
+            lastScope = getLastScopeFromProcessTerminator(mv, bb, funcName, labelGen, lastScope, visitedScopesSet);
```

Comment:
Maybe we might have to do the same thing in the InteropMethodGen as well? Can you write an interop function for the same usecase and verify and do the same thing there as well? Probably add this code to `JvmCodeGenUtil` and call in both places

Reply:
Yes, addressed with https://github.com/ballerina-platform/ballerina-lang/pull/31829/commits/2ce332c239451f46555c9590b3bf4311baf2a1cf